static/js/public/first-snap-flow.js

```diff
@@ -255,22 +257,66 @@ function initRegisterName(formEl, notificationEl, successEl) {
   formEl.addEventListener("submit", event => {
     event.preventDefault();
     let formData = new FormData(formEl);
+    const submitButton = formEl.querySelector(
+      "[data-js='js-snap-name-register']"
+    );
+
+    const currentPanel = formEl.closest(".p-accordion__group");
+    const currentToggle = currentPanel.querySelector(".p-accordion__tab");
+    const nextPanel = currentPanel.nextElementSibling;
+    const nextToggle = nextPanel.querySelector(".p-accordion__tab");
+
+    // Show spinner if data fetch takes long
+    const timer = setTimeout(() => {
+      if (currentToggle.getAttribute("aria-expanded") === "true") {
+        submitButton.classList.add("has-spinner");
+        submitButton.disabled = true;
+      }
+    }, 400);
 
     fetch("/register-snap/json", {
       method: "POST",
       body: formData
     })
       .then(response => response.json())
       .then(json => {
+        clearTimeout(timer);
         if (json.errors) {
```

Comment:
This could be reduced down:
```
clearTimeout(timer);
let message;
if (json.errors) {
  showError(json.errors[0].message);
  if (submitButton.disabled) {
    ...
  } 
  return;
} else if (json.code === "created") {
  message = `Name "${json.snap_name}" registered successfully.`;
} else if (json.code === "already_owned") {
  message = `You already own "${json.snap_name}"".`;
}
// Jump to the next accordion panel
toggleAccordion(currentToggle, false);
toggleAccordion(nextToggle, true);

if (submitButton.disabled) {
  submitButton.classList.remove("has-spinner");
  submitButton.disabled = false;
}

showSuccess(message); 
```